assignments/prototypes.js

```diff
@@ -32,16 +44,26 @@
   * should inherit destroy() from GameObject through CharacterStats
   * should inherit takeDamage() from CharacterStats
 */
+function Humanoid(attr) {
+  this.faction = attr.faction;
+  this.weapons = attr.weapons;
+  this.language = attr.language;
+  this.greet = () => `${attr.name} offers a greeting in ${attr.language}.`;
+  CharacterStats.call(this, attr);
```

Comment:
Ideally, you would want to do the call first and then assign the attributes. This will prevent any unintended side-effects in edge cases.